dev-tools/mage/gotest.go

```diff
@@ -193,8 +193,11 @@ func GoTest(ctx context.Context, params GoTestArgs) error {
 	args := []string{"test"}
 	args = append(args, "-v")
 
-	if params.Race {
-		args = append(args, "-race")
+	// -race is only supported on */amd64
+	if os.Getenv("DEV_ARCH") == "amd64" {
+		if params.Race {
+			args = append(args, "-race")
+		}
```

Comment:
I don't know if we could use a different approach. The current issue happens when running in a CI Windows worker the `mage dumpVariables` reports the environment variable GOARCH and GOOS accordingly, but when running in the same CI windows worker the command `mage -v build unitTest` those variables are pointing to something else `Adding build environment vars: map[CGO_ENABLED:0 GOARCH:amd64 GOFLAGS:-mod=vendor GOOS:linux]`